### ISSUE TITLE
[Feature Request] Toggle Dark Reader from command line

### ISSUE THREAD

**Issue by Hubro** [darkreader/darkreader#2870]
**Is your feature request related to a problem? Please describe.**

I am working on a script for automatically switching between light/dark themes across all my applications. This script saves me tons of time since I usually switch to a light theme, then back to a dark theme, at least twice per day, in several applications.

It would be amazing if I could toggle the Dark Reader plugin (on/off) from the command line in Linux, so it could be included in my global theme switching script.

**Describe the solution you'd like**

With my limited knowledge of extension development, I can imagine two possible solutions:
- Have an editable config file inside the extension directory that I could modify using (for example) `sed`, and make the extension poll that file regularly
- Provide an extension URL that I could open in my browser that would enable or disable the extension. This would be a less clean solution, as a new tab would need to be opened in the browser to toggle the extension, which means the browser would be focused

**Additional context**

I really love this extension and have been an avid user for years. From today forward I am also a monthly contributor :slightly_smiling_face:

**Comment by Gusted** [comment]
#1213

**Comment by WhyNotHugo** [comment]
I'd love to see this happen too.

I've written [darkman](https://gitlab.com/WhyNotHugo/darkman), which switched desktop theme to dark mode at sundown and back at sunrise.

I'd love to be able to signal darkreader somehow and tell it to switch modes.

There's work being done to expose a D-Bus service at an OS level. I believe it is possible for a Firefox add-on to listen to host D-Bus signals using [`nativeMessaging`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging). Would a PR for that be acceptable?

**Comment by Gusted** [comment]
> There's work being done to expose a D-Bus service at an OS level. I believe it is possible for a Firefox add-on to listen to host D-Bus signals using [`nativeMessaging`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging). Would a PR for that be acceptable?

#3519 This is currently the PR to add any external connections, it's been on a stale for a while due to some bugs that got around and other little roadmap things. I might squeeze it for this weekend to clean-up the PR a bit, as it should be a great feature to get Dark Reader integrated into other `eco-system`.

**Comment by WhyNotHugo** [comment]
This would allow for another extension to communicate with an external process that communicates with the system API to sync the dark-mode status, right?

That sounds like one hop too many; why rely on an external extension instead of implementing this into `darkreader`?

**Comment by Gusted** [comment]
> This would allow for another extension to communicate with an external process that communicates with the system API to sync the dark-mode status, right?

Yes, because we use a `passive` system into the PR, that just setup a listening port and let other extension send messages to it.
I didn't add any nativeMessaging support, because we as Dark Reader need to make the first message to a nativeMessage `host`, this can be done by having a list in Dark Reader that users to add nativeMessage and let Dark Reader deal with sending a `hello world` message to let the native application know it can go send options etc. However this also require intervals to reguarly check if any `inactive` native applications are finally `awake`. It requires some good logic and I prefer to implement this in another PR after the external connections is polish so I know what the `guidelines` are and how the logic definitely looks like.

**Comment by WhyNotHugo** [comment]
I understand, makes sense.

So will DarkReader eventually have `nativeMessaging` support itself, or will it rely on another extension to do that? (e.g.: a Linux-specific extension that check dark-mode setting via D-Bus and relays this to DarkReader).

I'm asking mostly to have a clearer direction of where this is headed and how to integrate best.

**Comment by Gusted** [comment]
> So will DarkReader eventually have `nativeMessaging` support itself, or will it rely on another extension to do that? (e.g.: a Linux-specific extension that check dark-mode setting via D-Bus and relays this to DarkReader).

The current PR only will rely on another extension, but can and will be expanded to have `nativeMessaging` Support. I actually think having an extra button under `Automation` to a new page with some settings should be understand-able and not too hard to squeeze into the current PR.
